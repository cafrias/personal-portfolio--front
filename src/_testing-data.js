export const PageTemplateProps = {
  inTransitionAnimation: false,
  setIsLast: jest.fn(),
  setNextPage: jest.fn(),
  setPreviousPage: jest.fn()
}
