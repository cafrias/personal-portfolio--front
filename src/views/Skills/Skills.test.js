import React from 'react'
import {shallow} from 'enzyme'

import {SkillsTesting} from './index'

import {PageTemplateProps} from '../../_testing-data'

describe('Skills', () => {
  const wrapper = shallow(<SkillsTesting {...PageTemplateProps} />)

  it('should render without crashing', () => {
    expect(wrapper).toHaveLength(1)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
