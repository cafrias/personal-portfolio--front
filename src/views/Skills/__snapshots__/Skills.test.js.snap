exports[`Skills should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <main
    className="page-template transition-animation ">
    <Connect(Component)
        direction="up"
        label="About"
        link="/about"
        ready={false} />
    <header
        className="section-header w-100">
        <PopIn
            onRest={[Function]}>
            <h1
                aria-label="Skills"
                className="section__heading">
                <img
                    alt="About"
                    className="img__heading"
                    src="heading.svg" />
            </h1>
        </PopIn>
    </header>
    <Skills
        ready={false}
        setIsLast={[Function]}
        setNextPage={[Function]}
        setPreviousPage={[Function]} />
    <Connect(Component)
        direction="down"
        label="Portfolio"
        link="/portfolio"
        ready={false} />
</main>,
  "nodes": Array [
    <main
      className="page-template transition-animation ">
      <Connect(Component)
            direction="up"
            label="About"
            link="/about"
            ready={false} />
      <header
            className="section-header w-100">
            <PopIn
                  onRest={[Function]}>
                  <h1
                        aria-label="Skills"
                        className="section__heading">
                        <img
                              alt="About"
                              className="img__heading"
                              src="heading.svg" />
                  </h1>
            </PopIn>
      </header>
      <Skills
            ready={false}
            setIsLast={[Function]}
            setNextPage={[Function]}
            setPreviousPage={[Function]} />
      <Connect(Component)
            direction="down"
            label="Portfolio"
            link="/portfolio"
            ready={false} />
</main>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <PageTemplate
        inTransitionAnimation={false}
        setIsLast={[Function]}
        setNextPage={[Function]}
        setPreviousPage={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": PageTemplate {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "inTransitionAnimation": false,
          "setIsLast": [Function],
          "setNextPage": [Function],
          "setPreviousPage": [Function],
        },
        "refs": Object {},
        "state": Object {
          "titleAnimationEnd": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <main
          className="page-template transition-animation ">
          <Connect(Component)
                    direction="up"
                    label="About"
                    link="/about"
                    ready={false} />
          <header
                    className="section-header w-100">
                    <PopIn
                              onRest={[Function]}>
                              <h1
                                        aria-label="Skills"
                                        className="section__heading">
                                        <img
                                                  alt="About"
                                                  className="img__heading"
                                                  src="heading.svg" />
                              </h1>
                    </PopIn>
          </header>
          <Skills
                    ready={false}
                    setIsLast={[Function]}
                    setNextPage={[Function]}
                    setPreviousPage={[Function]} />
          <Connect(Component)
                    direction="down"
                    label="Portfolio"
                    link="/portfolio"
                    ready={false} />
</main>,
        "_debugID": 2,
        "_renderedOutput": <main
          className="page-template transition-animation ">
          <Connect(Component)
                    direction="up"
                    label="About"
                    link="/about"
                    ready={false} />
          <header
                    className="section-header w-100">
                    <PopIn
                              onRest={[Function]}>
                              <h1
                                        aria-label="Skills"
                                        className="section__heading">
                                        <img
                                                  alt="About"
                                                  className="img__heading"
                                                  src="heading.svg" />
                              </h1>
                    </PopIn>
          </header>
          <Skills
                    ready={false}
                    setIsLast={[Function]}
                    setNextPage={[Function]}
                    setPreviousPage={[Function]} />
          <Connect(Component)
                    direction="down"
                    label="Portfolio"
                    link="/portfolio"
                    ready={false} />
</main>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PageTemplate
    inTransitionAnimation={false}
    setIsLast={[Function]}
    setNextPage={[Function]}
    setPreviousPage={[Function]} />,
}
`;
